src/shared/parse.h

```diff
@@ -54,4 +54,47 @@ int Parse_FreeSourceHandle(int handle);
 bool Parse_ReadTokenHandle(int handle, pc_token_t *pc_token);
 int Parse_SourceFileAndLine(int handle, char (&filename)[MAX_QPATH], int *line);
 
+/*
+===============
+Parse_CSV
+
+Tokenize a list like "a, b, c" to "a", "b", "c". This provides (part of) an
+input iterator over a string.
+
+This parser is quite tolerant: you can use one delimiter or another,
+or a variant of that: "a b,c ,d,,e    f" is equivalent to "a, b, c, d, e, f".
+
+Exemple usage:
+	for (Parse_CSV iterator(string); *iterator; ++iterator)
+	{
+		printf("C str: »%s«\n", *parser);
+	}
+===============
+*/
+// TODO in C++20: template<std::string delimiters = ", ">
+class Parse_CSV {
+public:
+	using value_type = char *;
+	using reference = char * const&;
+	using pointer   = char * const*;
+	using difference_type = std::ptrdiff_t;
+	using iterator_category = std::input_iterator_tag;
+
+	Parse_CSV(const std::string &);
+	Parse_CSV(const char *);
+	~Parse_CSV();
+
+	const char *operator*() const;
+
+	Parse_CSV& operator++(); // only ++i is implemented. there is no i++.
+
+private:
+	char *_string; // the string
```

Comment:
For some reason, I couldn't get the std::string version to modify things in-place. I suspect this would have to do with the `'\0'` in the string but I don't know

Reply:
There's no difference from how you do it with char*